src/modules/external_services/external_ip.py

```diff
@@ -0,0 +1,45 @@
+from .exceptions import ExternalServiceFailure, ExternalProviderFailure
+
+import requests
+from collections import namedtuple
+
+def generic_text_http_provider(url):
+    try:
+        return requests.get(url).text
+    except requests.ConnectionError as e:
+        raise ExternalProviderFailure(e)
+
+ExtIPProvider = namedtuple('ExternalIPProvider', [
+    'handler',
```

Comment:
Use more imperative name (e.g. `get`/`fetch`/`provide`) to make the usage more clear